includes/modules/extra_definitions.php

```diff
@@ -1,74 +1,48 @@
- */
-// must be called appropriately
-if (!defined('IS_ADMIN_FLAG')) {
-  die('Illegal Access');
-}
-// Set current template
-$template_id = $template_dir;
-
-// set directories to check for language files
-$languages_extra_definitions_directory = DIR_FS_CATALOG . DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/';
-$languages_extra_definitions_directory_template = DIR_FS_CATALOG . DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/' . $template_id . '/';
-
-$ws_languages_extra_definitions_directory = DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/';
-$ws_languages_extra_definitions_directory_template = DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/' . $template_id . '/';
-
-// Check for new definitions in template directory
-
-$directory_array = array();
-if ($dir = @dir($languages_extra_definitions_directory_template)) {
-  while ($file = $dir->read()) {
-    if (!is_dir($languages_extra_definitions_directory_template . $file)) {
-      if (preg_match('~^[^\._].*\.php$~i', $file) > 0) {
-        $directory_array[] = $file;
-      }
-    }
-  }
-  if (sizeof($directory_array)) {
-    sort($directory_array);
-  }
-  $dir->close();
-}
-
-// Check for new definitions in extra_definitions directory
-$dir_check = $directory_array;
-
-if ($dir = @dir($languages_extra_definitions_directory)) {
-  while ($file = $dir->read()) {
-    if (!is_dir($languages_extra_definitions_directory . $file)) {
-      if (preg_match('~^[^\._].*\.php$~i', $file) > 0) {
-        if (in_array($file, $dir_check, TRUE)) {
-          // skip name exists
-        } else {
-          $directory_array[] = $file;
-        }
-      }
-    }
-  }
-  if (sizeof($directory_array)) {
-    sort($directory_array);
-  }
-  $dir->close();
-}
-
-$file_cnt=0;
-for ($i = 0, $n = sizeof($directory_array); $i < $n; $i++) {
-  $file_cnt++;
-  $file = $directory_array[$i];
-
-  if (file_exists($ws_languages_extra_definitions_directory_template . $file)) {
-    include($ws_languages_extra_definitions_directory_template . $file);
-  } else {
-    include($ws_languages_extra_definitions_directory . $file);
-  }
-}
+ */
+// must be called appropriately
+if (!defined('IS_ADMIN_FLAG')) {
+  die('Illegal Access');
+}
+
+// set directories to check for language files
+$lang_extra_defs_dir = DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/';
+$lang_extra_defs_dir_template = DIR_WS_LANGUAGES . $_SESSION['language'] . '/extra_definitions/' . $template_dir . '/';
+
+$file_array = array(); 
+$folderlist = array($lang_extra_defs_dir_template, $lang_extra_defs_dir);
+
+foreach ($folderlist as $folder) { 
+  $this_folder = DIR_FS_CATALOG . $folder; 
+  if ($dir = @dir($this_folder)) {
+    while ($file = $dir->read()) {
+      if (!is_dir($this_folder. $file)) {
+        if (array_key_exists($file, $file_array)) {
+          // skip name exists
+        } else {
+          $file_array[$file] = $folder . $file;
```

Comment:
Without any additional test of `$file`, this will add all files that are in the directory including `.htaccess`, any hidden file and those that are not related to `.php` including files "renamed" to potentially prevent being processed.  Shouldn't an additional filter as provided in the removed code be included to eliminate files without an extension of `php`, files that begin with a period (`.`) and perhaps files that begin with an underscore (`_`)? (I assume the underscore option was added at some point to make it easy to "hide" a file from being loaded.)

Another thing, the use of the filename as the array key, doesn't this use more memory than a numerical index? The filename is or seems to already exist in the value and could be identified if needed, couldn't it?

Reply:
Added $file check - good catch.